src/javascript/EditPanelContainer/EditPanel/EditPanelContent/PreviewContainer/PreviewContainer.jsx

```diff
@@ -0,0 +1,83 @@
+import {Badge, Paper, Typography} from '@jahia/design-system-kit';
+import {ContentPreview} from '@jahia/react-apollo';
+import {PreviewComponent} from '@jahia/react-material';
+import {Grid, withStyles} from '@material-ui/core';
+import {connect} from 'formik';
+import * as PropTypes from 'prop-types';
+import React from 'react';
+import {compose} from 'react-apollo';
+import {translate} from 'react-i18next';
+import {setPreviewRefetcher} from '../../../EditPanel.refetches';
+
+const styles = theme => ({
+    contentPaper: {
+        display: 'flex',
+        flex: '1 1 auto',
+        flexDirection: 'column',
+        minWidth: 0
+    },
+    badge: {
+        margin: 0
+    },
+    preview: {
+        marginBottom: theme.spacing.unit * 2
+    }
+});
+
+const PreviewContainerCmp = ({t, classes, formik, editorContext}) => {
+    const workspace = 'EDIT';
+
+    return (
+        <Paper className={classes.contentPaper}>
+            <Grid container
+                  direction="row"
+                  justify="space-between"
+                  alignItems="center"
+                  className={classes.preview}
+            >
+                <Grid item>
+                    <Typography variant="epsilon" color="alpha">
+                        {t('content-editor:label.contentEditor.preview.toggleButtons.preview')}
+                    </Typography>
+                </Grid>
+
+                {formik.dirty &&
+                <Grid item>
+                    <Badge className={classes.badge}
+                           badgeContent={t('content-editor:label.contentEditor.preview.updateOnSave')}
+                           variant="normal"
+                           color="info"
+                    />
+                </Grid>
+                }
+            </Grid>
+
+            <ContentPreview path={editorContext.path}
+                            language={editorContext.lang}
+                            workspace={workspace}
+                            templateType="html"
+                            view="cm"
+                            contextConfiguration="preview"
+                            fetchPolicy="network-only"
+                            setRefetch={refetchingData => setPreviewRefetcher(refetchingData)}
+            >
+                {data => <PreviewComponent data={data.jcr ? data.jcr : {}} workspace={workspace}/>}
+            </ContentPreview>
+        </Paper>
+    );
+};
+
+PreviewContainerCmp.propTypes = {
+    editorContext: PropTypes.object.isRequired,
+    t: PropTypes.func.isRequired,
+    classes: PropTypes.object.isRequired,
+    formik: PropTypes.object.isRequired
+};
+
+const PreviewContainer = compose(
+    translate(),
+    withStyles(styles),
+    connect
+)(PreviewContainerCmp);
+
+export {PreviewContainer};
```

Comment:
```suggestion
PreviewContainer.displayName = 'PreviewContainer';
```

It's help a lot for debugging with hoc

Reply:
Can you give us any article or documentation related to this topic !

Reply:
I think it's because of the HoC wrapping the names.
if you look at the name of component wrapped by HoC it will be: withStyles(PreviewContainer)
I read some topics: https://www.freecodecamp.org/news/10-points-to-remember-thatll-help-you-master-coding-in-reactjs-library-d0520d8c73d8/

And it seem's it a best practice to define the displayName for debugging using react developper tools. May be we should consider adding a lint rule to check that components have display names, but we could leave this discussion to the front guild.

For now, I agree with Florent on this.

Reply:
Also you have top accept this modification, because currently there is 2 export that are conflicting.

Reply:
https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging

 :-)

Reply:
thank you

Reply:
```
Also you have to accept this modification, because currently there is 2 export that are conflicting.
```

Conflicting export due to the 1st and 2nd suggestions and not to the implementation.